tests/patmat/gadt2.scala

```diff
@@ -0,0 +1,16 @@
+sealed trait Nat[+T]
+case class Zero() extends Nat[Nothing]
+case class Succ[T]() extends Nat[T]
```

Comment:
This is not intended to be a correct representation of Peano numerals, right? Not only for the `+T` but also because `T` in `Succ` is completely unconstrained...